homeassistant/components/hko/weather.py

```diff
@@ -0,0 +1,83 @@
+"""Support for the HKO service."""
+from homeassistant.components.weather import (
+    Forecast,
+    WeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LOCATION, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    API_CONDITION,
+    API_CURRENT,
+    API_FORECAST,
+    API_HUMIDITY,
+    API_TEMPERATURE,
+    ATTRIBUTION,
+    DOMAIN,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a HKO weather entity from a config_entry."""
+    name = config_entry.data[CONF_LOCATION]
+    unique_id = config_entry.unique_id
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([HKOEntity(name, unique_id, coordinator)], False)
+
+
+class HKOEntity(CoordinatorEntity, WeatherEntity):
```

Comment:
Please read up on [entity naming rules](https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations), adjust if needed (it looks good to me, I don't think any change is needed), and set the `_attr_has_entity_name` flag to True.

Reply:
Also include `DeviceInfo` (service)